Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -1307,41 +1421,73 @@ protected async ValueTask<ItemsProviderResult<TItem>> VirtualizeItemsProviderHan
+        bool suppressSortOrderChangedEvent )
     {
-        if ( Sortable && column.CanSort() )
+        if ( !Sortable || !column.CanSort() ) return;
+
+        if ( SortMode == DataGridSortMode.Single )
         {
-            if ( SortMode == DataGridSortMode.Single )
+            // in single-mode we need to reset all other columns to default state
+            foreach ( var c in Columns.Where( x => x.GetFieldToSort() != column.GetFieldToSort() ) )
             {
-                // in single-mode we need to reset all other columns to default state
-                foreach ( var c in Columns.Where( x => x.GetFieldToSort() != column.GetFieldToSort() ) )
-                {
-                    c.CurrentSortDirection = SortDirection.Default;
-                }
-
-                // and also remove any column sort info except for current one
-                SortByColumns.RemoveAll( x => x.GetFieldToSort() != column.GetFieldToSort() );
+                c.CurrentSortDirection = SortDirection.Default;
             }
 
-            if ( changeSortDirection )
-            {
-                column.CurrentSortDirection = sortDirection ?? column.CurrentSortDirection.NextDirection( column.ReverseSorting );
-            }
+            // and also remove any column sort info except for current one
+            SortByColumns.RemoveAll( x => x.GetFieldToSort() != column.GetFieldToSort() );
+        }
 
-            if ( !SortByColumns.Any( c => c.GetFieldToSort() == column.GetFieldToSort() ) )
-            {
-                var nextOrderToSort = SortByColumns.Count == 0 ? 0 : SortByColumns.Max( x => x.SortOrder ) + 1;
-                column.SetSortOrder( nextOrderToSort );
-                SortByColumns.Add( column );
-            }
-            else if ( column.CurrentSortDirection == SortDirection.Default )
-            {
-                SortByColumns.Remove( column );
-                column.ResetSortOrder();
-            }
+        if ( changeSortDirection )
+        {
+            column.CurrentSortDirection =
+                sortDirection ?? column.CurrentSortDirection.NextDirection( column.ReverseSorting );
+        }
 
-            if ( changeSortDirection )
-                InvokeAsync( () => SortChanged.InvokeAsync( new DataGridSortChangedEventArgs( column.GetFieldToSort(), column.CurrentSortDirection ) ) );
+        if ( SortByColumns.All( c => c.GetFieldToSort() != column.GetFieldToSort() ) )
+        {
+            var nextOrderToSort = SortByColumns.Count == 0 ? 0 : SortByColumns.Max( x => x.SortOrder ) + 1;
+            column.SetSortOrder( nextOrderToSort );
+            SortByColumns.Add( column );
+        }
+        else if ( column.CurrentSortDirection == SortDirection.Default )
+        {
+            SortByColumns.Remove( column );
+            column.ResetSortOrder();
+        }
+
+        Task RaiseSortChanged() =>

```

Comment:
Are these local functions here? Aren't we capturing variables needlessly?

Reply:
Changed the local function to `static` in bf531e2209cfcf20d3ec1467fb1adf5c272dbb90.